### PR TITLE
Allow users to add new stages without having to reload the edit all page and losing their uunsaved data

### PR DESCRIPTION
**Pull request by yanndago**
- Modify the "Add stage dialog" to receive a custom submit handler that replaces the default behavior of creating a new stage.
- Modify the edit all page to add a stage on the client only via a custom submit handler that it passes to the "Add stage dialog"
- Modify the form field names of stages added only on the client so that they can be created on submit
- Modify the server so that it creates new stages for the stages that were only created on the client

### CODE REVIEW

**Comment by jrobbins** [comment]
I'll defer to Daniel's review on this PR because he has been working more closely in this area of the code.

**Review by DanielRyanSmith — COMMENTED** [review]
I don't know if someone has requested this to land soon, but I'm wondering if we can hold off on this change for a while until the changes to phase out the code in `guide.py` have been landed, which should be very soon. This adds a lot of complexity to the old submission process which is being overhauled, and this work will need to be translated to fit that. Sorry for the collision here; I was not aware this was something you were working on.

**Review by DanielRyanSmith — APPROVED** [review]
I tried this out locally and it seems to function just as intended, even when adding multiple stages in the same edit. 🙂

**Pull request merged**
